### PR TITLE
fix(Button): Text color in HC mode for hover, active and checked states

### LINKED ISSUE #22464: [Bug]: Button V9 Teams theme: High contrast text color is white, should be black to meet accessibility requirements
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
v9
```

### Are you reporting Accessibility issue?

no

### Reproduction

https://6002298f95a00c00213f4d55-qpafvorpoj.chromatic.com/?path=/docs/components-button--default

## Actual Behavior
color is white

## Expected Behavior
color should be black

### Logs

_No response_

### Requested priority

High

### Products/sites affected

_No response_

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by jurokapsiar**
Fixes #22464, part of #21998

This PR fixes the text color for high contrast themes in the hover and active states for all `Buttons`, and in the checked states as well for `ToggleButtons`.

### CODE REVIEW

File: packages/react-components/src/DocsComponents/ThemePicker.stories.tsx [hunk 1]
```diff
@@ -34,24 +44,32 @@ export const ThemePicker: React.FC<{ selectedThemeId?: string }> = ({ selectedTh
...
+          <MenuList>
+            {themes.map(theme => (
+              <MenuItemRadio name="theme" value={theme.id} key={theme.id}>
+                {theme.label}
+              </MenuItemRadio>
+            ))}
+          </MenuList>
+        </MenuPopover>
+      </Menu>
+      <Button>Default</Button>
```
Comment (jurokapsiar) [inline packages/react-components/src/DocsComponents/ThemePicker.stories.tsx:68]: temporary as theme switching did not work for me :(

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                                                                                             |      Baseline (minified/GZIP) |                           PR |                                                                                                                                                                                                Change |
| :---------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: | ---------------------------: | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/Button.fixture.js'>Button</abbr>                                                                                    |    `28.083 kB`<br />`8.09 kB` |  `28.088 kB`<br />`8.093 kB` |  `5 B` <br />`3 B`  |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/CompoundButton.fixture.js'>CompoundButton</abbr>                                                                    |   `33.578 kB`<br />`9.126 kB` |  `33.583 kB`<br />`9.128 kB` |  `5 B` <br />`2 B`  |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/MenuButton.fixture.js'>MenuButton</abbr>                                                                            |    `29.85 kB`<br />`8.684 kB` |  `29.855 kB`<br />`8.687 kB` |  `5 B` <br />`3 B`  |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/SplitButton.fixture.js'>SplitButton</abbr>                                                                          |   `36.322 kB`<br />`9.878 kB` |  `36.327 kB`<br />`9.881 kB` |  `5 B` <br />`3 B`  |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/ToggleButton.fixture.js'>ToggleButton</abbr>                                                                        |   `37.465 kB`<br />`8.717 kB` |   `37.47 kB`<br />`8.731 kB` | `5 B` <br />`14 B`  |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `174.881 kB`<br />`49.118 kB` | `174.886 kB`<br />`49.12 kB` |  `5 B` <br />`2 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                  |         Size (minified/GZIP) |
| -------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------: |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/Card.All.fixture.js'>Card - All</abbr>                                                     | `57.884 kB`<br />`16.524 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/Card.fixture.js'>Card</abbr>                                                               | `53.169 kB`<br />`15.238 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardFooter.fixture.js'>CardFooter</abbr>                                                   |   `7.756 kB`<br />`3.299 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardHeader.fixture.js'>CardHeader</abbr>                                                   |   `9.321 kB`<br />`3.816 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardPreview.fixture.js'>CardPreview</abbr>                                                 |   `7.728 kB`<br />`3.323 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr> | `32.671 kB`<br />`10.696 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/43f3fbf79244dfe2557322b1418f73d93bd1b054'>43f3fbf79244dfe2557322b1418f73d93bd1b054</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22566/perf-test-react-components/Avatar-mount_base.html">1278</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22566/perf-test-react-components/Avatar-mount.html">1314</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22566/perf-test-react-components/Button-mount_base.html">773</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22566/perf-test-react-components/Button-mount.html">770</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22566/perf-test-react-components/FluentProvider-mount_base.html">2213</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22566/perf-test-react-components/FluentProvider-mount.html">2191</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22566/perf-test-react-components/FluentProviderWithTheme-mount_base.html">433</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22566/perf-test-react-components/FluentProviderWithTheme-mount.html">407</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22566/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">362</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22566/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">351</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22566/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">446</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22566/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">437</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22566/perf-test-react-components/MakeStyles-mount_base.html">1969</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22566/perf-test-react-components/MakeStyles-mount.html">1938</a></td>
            <td>50000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by ling1726** [comment]
If this is not breaking a screener test then we should add a screener test :)

**Review by khmakoto — APPROVED** [review]

**Comment by jurokapsiar** [comment]
thank you @khmakoto - @ling1726 fyi: 
image

**Pull request merged**